### PR TITLE
[filesize] Update to 3.6.0

### PR DESCRIPTION
**Pull request by renchap**
Updating to the latest version.
Options have been updated according to the README documentation, added the `partial` function, updated tests with the latest examples from the README.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/avoidwork/filesize.js/blob/master/README.md
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/filesize/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-// Type definitions for filesize 3.2.1
+// Type definitions for filesize 3.6.0
```
Comment (DanielRosenwasser) [inline types/filesize/index.d.ts]: Looks like our lint rules are disabled for this package. Can you enable the lints in our `tslint.json` and fix up the version string to only be a `MAJOR.MINOR` version (e.g. `4.7`), as opposed to including the patch or containing a leading `v` (e.g. `v1.2.3`).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/filesize/index.d.ts [hunk 2]
```diff
@@ -36,47 +37,64 @@ declare namespace Filesize {
...
+        * Output of function (array, exponent, object, or string), default is string
+        */
+        output?: "array" | "exponent" | "object" | "string";
         /**
         * Decimal place, default is 2
         */
         round?: number;
         /**
-        * Output of function (array, exponent, object, or string), default is string
+        * Decimal separator character, default is .
```
Comment (DanielRosenwasser) [inline types/filesize/index.d.ts]: Probably want to put this in backticks or quotemarks

**Comment by renchap** [comment]
Thanks for the review @DanielRosenwasser 
I do not know why but a lot of TSLint rules were ignored in `tslint.json`. Now it only extends `dtslint/dt.json` and I fixed all the linting errors.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thank you very much!

**Pull request merged**